lib/features/profiles/screens/profile_selection_screen.dart

```diff
@@ -0,0 +1,196 @@
+// ignore_for_file: prefer_const_constructors, prefer_const_literals_to_create_immutables
+
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:givt_app_kids/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app_kids/features/auth/screens/login_screen.dart';
+import 'package:givt_app_kids/features/profiles/cubit/profiles_cubit.dart';
+import 'package:givt_app_kids/features/profiles/models/profile.dart';
+import 'package:givt_app_kids/features/profiles/widgets/profile_item.dart';
+
+import 'package:givt_app_kids/helpers/analytics_helper.dart';
+
+class ProfileSelectionScreen extends StatefulWidget {
+  static const String routeName = "/profile-selection-bloc";
+
+  const ProfileSelectionScreen({Key? key}) : super(key: key);
+
+  @override
+  State<ProfileSelectionScreen> createState() => _ProfileSelectionScreenState();
+}
+
+class _ProfileSelectionScreenState extends State<ProfileSelectionScreen> {
+  @override
+  void initState() {
+    super.initState();
+    _fetchProfiles();
+  }
+
+  Future<void> _fetchProfiles() async {
+    final parentGuid = (context.read<AuthCubit>().state as LoggedInState).guid;
+    context.read<ProfilesCubit>().fetchProfiles(parentGuid);
+  }
+
+  Future<void> _selectProfile(Profile profile) async {
+    context.read<ProfilesCubit>().setActiveProfile(profile);
+    await AnalyticsHelper.logEvent(
+      eventName: AmplitudeEvent.profilePressed,
+      eventProperties: {
+        "profile_name": '${profile.firstName} ${profile.lastName}',
+      },
+    );
+  }
```

Comment:
Wouldn't be better to use only the Profile id to set it to active? Instead of the whole Profile object. Usually you don't want to use in the events data that is already in the BLoC state

Reply:
yeah, I also thought about storing either id or index of profiles, but implemented as is just to keep some time (I reused existing code when it was possible). But it's a good point for improvement, thanks ;)